event-sourcing/event-source/src/test/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceProducerTest.java

```diff
@@ -44,16 +43,38 @@
     private EventSourceProducer eventSourceProducer;
 
 
+    @Test
+    public void shouldCreateDefaultEventSourceDefinitionWhenEventSourceNameIsEmpty() throws Exception {
+        final EventSourceDefinition eventSourceDefinition = eventSourceDefinition()
+                .withName("defaultEventSource")
+                .withDefaultEventSource(true)
+                .withLocation(new Location("", "", Optional.of("dataSource")))
+                .build();
+        final InjectionPoint injectionPoint = mock(InjectionPoint.class);
+        final EventSourceName eventSourceNameAnnotation = mock(EventSourceName.class);
+
+        final JdbcBasedEventSource jdbcBasedEventSource = mock(JdbcBasedEventSource.class);
+        when(qualifierAnnotationExtractor.getFrom(injectionPoint, EventSourceName.class)).thenReturn(eventSourceNameAnnotation);
+        when(eventSourceNameAnnotation.value()).thenReturn("");
+        when(eventSourceDefinitionRegistry.getDefaultEventSourceDefinition()).thenReturn(eventSourceDefinition);
+        when(jdbcEventSourceFactory.create(eventSourceDefinition.getLocation().getDataSource().get(), eventSourceDefinition.getName())).thenReturn(jdbcBasedEventSource);
+
+        assertThat(eventSourceProducer.eventSource(), is(jdbcBasedEventSource));
+    }
+
     @Test
     public void shouldCreateEventSourceUsingTheJNDINameInjectedByTheContainer() throws Exception {
```

Comment:
The test name does not seem to correspond with what is occurring in the test

Reply:
Test name changed